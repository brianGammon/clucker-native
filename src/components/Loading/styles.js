import { StyleSheet } from 'react-native';
import variables from 'native-base/dist/src/theme/variables/platform';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  message: {
    color: variables.inputColorPlaceholder,
  },
});
