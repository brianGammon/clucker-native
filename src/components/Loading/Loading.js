import * as React from 'react';
import { Spinner, Text, Container } from 'native-base';

import styles from './styles';

type Props = {
  message: string,
};

export default ({ message }: { message: Props.message }) => (
  <Container style={styles.container}>
    <Spinner color="rgb(249, 149, 0)" />
    {message && <Text style={styles.message}>{message}</Text>}
  </Container>
);
