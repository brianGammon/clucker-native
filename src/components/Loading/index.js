import React from 'react';
import {
  ActivityIndicator, StatusBar, View, Text,
} from 'react-native';

import styles from './styles';

type Props = {
  message: string,
};

export default ({ message }: { message: Props.message }) => (
  <View style={styles.container}>
    <ActivityIndicator size="large" />
    <StatusBar barStyle="default" />
    {message && <Text>{message}</Text>}
  </View>
);
