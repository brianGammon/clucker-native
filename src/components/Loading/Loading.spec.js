import React from 'react';
import Loading from '.';

describe('HeaderProfile component:', () => {
  test('Should render without message', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render with message', () => {
    const wrapper = shallow(<Loading message="Test message" />);
    expect(wrapper).toMatchSnapshot();
  });
});
