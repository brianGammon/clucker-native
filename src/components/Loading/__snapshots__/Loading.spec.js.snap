// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderProfile component: Should render with message 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    size="large"
  />
  <StatusBar
    animated={false}
    barStyle="default"
    showHideTransition="fade"
  />
  <Component>
    Test message
  </Component>
</Component>
`;

exports[`HeaderProfile component: Should render without message 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    size="large"
  />
  <StatusBar
    animated={false}
    barStyle="default"
    showHideTransition="fade"
  />
</Component>
`;
